PFW-1174: Add pause/resume for USB/host prints via LCD menu

3d-gussner opened the pull request:
Depending if SD or USB/host print the firmware sends
- SD print: `// action:paused` or `// action:resumed` are send to **inform** USB/Host
- USB/host print: `// action:pause` or `// action:resume` are send to trigger the USB/host to handle it

- USB/host must handle `// action:pause` and `// action:resume` correctly to work
  - Tested with Octoprint
    - It handles every thing correctly
    - Any combination of Octoprint and/or LCD `pause` and `resume` working correctly
  - Tested with Pronterface
    - It pauses BUT doesn't send the printer in pause position, and so it is not possible to `resume` from LCD menu
    - I guess some Macros can fix that.
  - Repetier Host/Server documentation shows that it should work. Not tested.

Could save 56 bytes in first step and additional 38 bytes adding `MSG_PAUSE_PRINT` to messages.c/.h

Updated `lang_en*.txt`
@todo Polish translation is 19 characters long (it still fits) BUT should be corrected to 18 chars.

3d-gussner committed:
Add pause/resume to USB/host prints via LCD menu

Depending if SD or USB/host print the firmware sends
- SD print: `// action:paused` or `// action:resumed` are send to inform USB/Host
- USB/host print: `// action:pause` or `// action:resume` are send to trigger the USB/host to handle it

- USB/host must handle `// action:pause` and `// action:resume` correctly to work
  - Tested with Octoprint
    - It handles every thing correctly
    - Any combination of Octoprint and/or LCD `pause` and `resume` working correctly
  - Tested with Pronterface
    - It pauses BUT doesn't send the printer in pause position, and so it is not possible to `resume` from LCD menu
    - I guess some Macros can fix that.
  - Repetier Host/Server documentation shows that it should work. Not tested.

Could save 56 bytes in first step and additional 38 bytes adding `MSG_PAUSE_PRINT` to messages.c/.h

Updated `lang_en*.txt`
@todo Polish translation is 19 characters long (it still fits) BUT should be corrected to 18 chars.

DRracer commented:
@3d-gussner looks correct to me, however we must retest the complete setup of Pause+Stop+Fan checks both SD print and Octoprint

DRracer commented on Firmware/ultralcd.cpp:
```diff
@@ -6631,44 +6643,50 @@ static void lcd_main_menu()
-		if (mesh_bed_leveling_flag == false && homing_flag == false) {
-			if (card.sdprinting)
+	#ifdef FANCHECK
+		if((fan_check_error == EFCE_FIXED) || (fan_check_error == EFCE_OK))
```
I'd wrap just the `if((fan_check_error == EFCE_FIXED) || (fan_check_error == EFCE_OK))` into `#ifdef FANCHECK` to avoid having the block body twice in the code:

```
#ifdef FANCHECK
		if((fan_check_error == EFCE_FIXED) || (fan_check_error == EFCE_OK))
#endif
        {
			if (is_usb_printing){
				MENU_ITEM_SUBMENU_P(_T(MSG_RESUME_PRINT), lcd_resume_usb_print);////MSG_RESUME_PRINT c=18
			} else {
				MENU_ITEM_SUBMENU_P(_T(MSG_RESUME_PRINT), lcd_resume_print);////MSG_RESUME_PRINT c=18
            }
        }
```

3d-gussner replied:
Pushed change

DRracer commented on Firmware/ultralcd.cpp:
```diff
@@ -6631,44 +6643,50 @@ static void lcd_main_menu()
-#else
-      MENU_ITEM_SUBMENU_P(_T(MSG_RESUME_PRINT), lcd_resume_print);////MSG_RESUME_PRINT c=18
-#endif
+	if (IS_SD_PRINTING)
```
the code indentation looks broken - either copy the style around or use spaces on the whole new functionality

3d-gussner replied:
Sorry my VScode had for some reason mix of tabs vs spaces using tabs.
Changed it to tab = 4 spaces in the function `lcd_main_menu`

DRracer reviewed (changes requested):
Please address my comments in the code

3d-gussner committed:
Indentation to 4 spaces for tabs

3d-gussner committed:
avoid having the block body twice in the code. Thanks to @DRracer

3d-gussner committed:
Fix NO Stop print during MBL

3d-gussner committed:
Documentation Show Main Menu

3d-gussner committed:
Fix Fan error issues.

3d-gussner committed:
Indentations

3d-gussner committed:
Don't show Settings during pause

3d-gussner committed:
Fix FANCHECK build error

3d-gussner committed:
Indentations

3d-gussner committed:
Merge remote-tracking branch 'upstream/MK3' into PFW-1174

Fix merge issues

3d-gussner committed:
Show LCD Settings during pause

3d-gussner committed:
Revert `M602`

3d-gussner committed:
Fix USB/host FAN Error resume
Show Settings during Pause also for USB/host prints

3d-gussner committed:
Again revert Settings

3d-gussner committed:
`|| isPrintPaused` is already in `PRINTER_ACTIVE`

3d-gussner committed:
Add "Fan check [On|Off]" menu to tune
User can en/disable the fan check during print. This is very useful in case of false positive
fan errors to let the user to decide to finish print with "faulty" fan.

DRracer commented:
tests are green, approving, ready to merge.
